src/yunohost/hook.py

```diff
@@ -388,11 +392,27 @@ def hook_exec(path, args=None, raise_on_error=False, no_trace=False,
                 errno.EIO, m18n.n('hook_exec_not_terminated', path=path))
         else:
             logger.error(m18n.n('hook_exec_not_terminated', path=path))
-            return 1
+            return 1, ''
     elif raise_on_error and returncode != 0:
         raise MoulinetteError(
             errno.EIO, m18n.n('hook_exec_failed', path=path))
-    return returncode
+
+    try:
+        with open(stdreturn, 'r') as f:
+            if f.read() != '':
+                returnjson = read_json(stdreturn)
+            else:
+                returnjson = {}
+    except Exception as e:
+        returnjson = {}
+        raise MoulinetteError(
+            errno.EIO, m18n.n('hook_json_return_error', path=path, msg=str(e)))
+
+    stdreturndir = os.path.split(stdreturn)[0]
+    os.remove(stdreturn)
+    os.rmdir(stdreturndir)
```

Comment:
Those operation won't be executed in case there is a raise line 408 thus letting the FS dirty.

Reply:
Should be fixed